src/features/postsSlice.ts

```diff
@@ -0,0 +1,54 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+export interface PostsState {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+// Асинхронный thunk для загрузки постов пользователя
+export const fetchUserPosts = createAsyncThunk(
+  'posts/fetchUserPosts',
+  async (userId: number) => {
+    const response = await client.get<Post[]>(`/posts?userId=${userId}`);
+
+    return response;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(fetchUserPosts.pending, state => {
+        // eslint-disable-next-line no-param-reassign
+        state.loaded = false;
+        // eslint-disable-next-line no-param-reassign
+        state.hasError = false;
+      })
+      .addCase(fetchUserPosts.fulfilled, (state, action) => {
+        // eslint-disable-next-line no-param-reassign
+        state.items = action.payload;
+        // eslint-disable-next-line no-param-reassign
+        state.loaded = true;
```

Comment:
Code quality issue: Similar to the previous comment, you're disabling the linter rule and mutating state directly. Consider creating a new state with the spread operator.